Add Security Policy and Community Health files (#365)
mautic/docker-mautic: SECURITY.md

# Mautic's Docker Security Policy

## Goals of the Mautic Docker Image Security Policy
- Maintain a secure official Docker image for Mautic
- Resolve reported security issues specific to the Docker image configuration
- Provide clear documentation on securely deploying and operating the Mautic Docker image
- Establish a transparent process for handling Docker-specific security vulnerabilities

## Scope of Mautic's Docker Security Team
The Mautic Docker Security Team's scope is limited to security issues specific to the official Mautic Docker image, including:

- Docker image configuration

> - Base image vulnerabilities

Reviewer: ```suggestion
- PHP secure configuration best practices
```

Base image vulnerabilities are out of our scope, because we can't really control that appart from updating to the latest version.

- Dockerfile security best practices
- Container runtime security concerns
- Docker-specific deployment guidance

The team does not directly handle:

- Mautic core application vulnerabilities (these are handled by the Mautic Security Team and should be raised [here](https://github.com/mautic/mautic/security/advisories/new))
- Third-party plugins not included in the official image
- Custom Docker deployments modified from the official image

> - Individual user deployment configurations

Reviewer: ```suggestion
- Individual user deployment configurations
- Base PHP image vulnerabilities
```

Author: I think this is not needed, because you're suggesting to add back in a line which is removed?

Reviewer: Actually I'm trying to add the "Base PHP image vulnerabilities", but I still dunno how to use this feature correctly lol.

Anyway, the section bellow is related to this topic, so I guess we don't need it.

> 

Reviewer: ```suggestion

## How are vulnerabilities in the underlying base image handled?
Mautic's Docker image is automatically rebuilt every Monday at 00:00 UTC to incorporate the latest security updates from the base image and any updated Debian packages, ensuring that known vulnerabilities are addressed regularly.

```

- Development, alpha, beta, or release candidate Docker images will not receive security advisories
- Docker images for unsupported Mautic versions will not receive security updates

## How to report a Docker security issue

If you discover a potential security vulnerability specific to the Mautic Docker image:

1. Keep it confidential - Do not discuss it publicly in issues, pull requests, forums, or Slack
2. Submit your concern as a private disclosure via GitHub's Security Advisory feature at https://github.com/mautic/docker/security
3. Provide detailed information about the vulnerability, including steps to reproduce and potential impact
4. If possible, you may create a private fork to propose a fix

## How Docker security issues are resolved

Mautic's Docker Security Team follow the [same process as the Mautic Security Team](https://mautic.org/security/how-security-issues-are-resolved/) when resolving issues.

1. The Security Team will triage incoming reports to determine validity and severity
2. Valid issues will be acknowledged within 24 hours and triaged within 7 business days
3. The team will develop and test fixes for confirmed vulnerabilities within 21 business days

> 4. Security fixes will be integrated into new Docker image releases on a quarterly basis, with the middle month in the quarter being security releases

Reviewer: Security fixes will be integrated into new Docker image builds, triggered as soon as the patch is merged into the corresponding production branch.

Reviewer: > I strongly suggest that we do security releases quarterly, the middle month of the quarter, alongside the Mautic security releases. Otherwise businesses cannot know when to expect them and prepare for them if they come piecemeal as-and-when we merge. I would recommend that we align timelines and only merge those fixes/release those builds at those times.

Hey Ruth,

I agree that we could do it on a scheduled basis, but I disagree that they should be synced with Mautic releases. 

A Docker package is, simplifying, a server on a package, making it prone to the same vulns as a regular server - so it should be patched at the same speed as any server.

I'd recommend rebuilds every 7 or 15 days so that the system keeps updated. 

Please refer to [Docker official best practices](https://docs.docker.com/build/building/best-practices/#rebuild-your-images-often).

We could automate a pipeline that rebuilds every monday, for example.

Reviewer: ```suggestion
4. Security patches will be integrated into new Docker image builds on the 2nd and 4th Wednesdays of each month, if applicable
```

> 5. Where an urgent fix is required, an out-of-cycle release is made in collaboration with the Mautic Security Team

Reviewer: Remove, according to my comment on the line above.

6. Coordination with the Mautic Security Team will occur when issues overlap with core application security

## Security fix announcements and releases

> - Security fixes will be announced alongside new Docker image releases

Reviewer: "new Docker image builds"

Author: You can use the 'suggest change' button to write what you think is better, then the PR author can just click to apply your suggestions.

![image](https://github.com/user-attachments/assets/9523f847-5780-4c14-9d2a-b3a35218f442)

Reviewer: ```suggestion
- Security fixes will be announced alongside new Docker image builds
```

> - Critical vulnerabilities may prompt immediate out-of-sequence releases

Reviewer: Remove, refer to line 47

> - Announcements will include the severity, affected versions, and remediation steps

Reviewer: ```suggestion
- Announcements will include the severity, affected builds, and remediation steps
```
